st2client/st2client/commands/inquiry.py

```diff
@@ -127,19 +127,29 @@ def __init__(self, resource, *args, **kwargs):
     def run(self, args, **kwargs):
         instance = Inquiry()
         instance.id = args.id
+        inquiry = self.get_resource_by_id(id=args.id, **kwargs)
         if args.response:
             instance.response = json.loads(args.response)
         else:
-            inquiry = self.get_resource_by_id(id=args.id, **kwargs)
             response = InteractiveForm(
                 inquiry.schema.get('properties')).initiate_dialog()
             instance.response = response
 
-        return self.manager.update(instance, **kwargs)
+        # Send the response to the API first, then sanitize secrets
+        instance = self.manager.update(instance, **kwargs)
```

Comment:
Hm, I'm a bit confused - shouldn't the masking happen on the server (API) side otherwise it kinda defeats the purpose?

That's how we do it for other endpoints. In addition to that, we support `?show_secrets` query param which can be used by admin to get the response back unmasked.

Reply:
Edit: I needed to expand the diff a bit to see more context.

The actual response and potential secrets are provided by the user and sent to the API (and not just returned to the user by the API) so the sanitization for `response` command is correct :)

And as far as get one and get all API endpoints go - we don't actually return the response provided by the user and which is stored as liveaction result as part of those APIs so we should be fine, right?

Reply:
Actually I didn't think through that well enough. For some reason I "remembered" that we aren't storing the response so didn't need to worry about it.....but of course this is wrong because we obviously need to use the response data for later tasks in the workflow.

This level of masking on the client-side, however, isn't wasted work - should still be done. I just need to continue on the server side.